pkg/server/proto/server.proto

```diff
@@ -392,6 +392,22 @@ service Waypoint {
   // based on the trigger configuration.
   rpc RunTrigger(RunTriggerRequest) returns (RunTriggerResponse);
 
+  //----------------------------------------------------------------------
+  // Task functions
+  //----------------------------------------------------------------------
+
+  // UpsertTask updates or inserts a Task
+  rpc UpsertTask(UpsertTaskRequest) returns (UpsertTaskResponse);
+
+  // GetTask returns a requested Task message. Or an error if it does not exist.
+  rpc GetTask(GetTaskRequest) returns (GetTaskResponse);
+
+  // DeleteTask takes a Task id and deletes it, if it exists.
+  rpc DeleteTask(DeleteTaskRequest) returns (google.protobuf.Empty);
```

Comment:
DeleteTask probably shouldn't exist? I don't think we want anyway to delete task metadata, similar to how we do not have a `DeleteJob`. We should instead have a `CancelTask` or something (if anything).

Reply:
I'll remove `DeleteTask` and wait to implement `CancelTask` once I'm through the Phase 3.1! Just pushed up a commit to remove the Delete API endpoint.